.github/workflows/links.yml

```
@@ -0,0 +1,21 @@
+name: links
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  check-links:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v3
```

Do we want to pin the minor and patch version?

---

This is how it is done [elsewhere](https://github.com/open-telemetry/opentelemetry-go/blob/587437ba5b784c4ca242a8374758d0ac37fa6bd8/.github/workflows/links.yml#L15). I think it is find to pin to a major version here. All we are looking for is to ensure backwards compatibility and have the thing run.

---

I experienced a few times issues that using newer non-major GitHub Action versions were causing issues (e.g. flaky builds) and using a not pinned can make it harder to find out that the issue was related to a bug introduced in a GitHub Action.  From my experience pinning dependencies can sometimes save hours when retrospecting a problem.

We have dependabot configured for GitHub Actions so pinning a version should not be a problem while we will have more "control" on the build pipeline.

On the other hand, I have never encountered an issue like that with `actions/checkout` 😉 Still at least the maintainers get a chance to know what are the new features/bugfixes thanks to the dependabot notifications.

 This comment is not a blocker for me so feel free to resolve if you have other opinion.

---

```suggestion
    - uses: actions/checkout@v3.1.0
```

if you decide to pin the version 😉